src/protocols/secure_channel/PASESession.cpp

```diff
@@ -482,38 +552,44 @@ CHIP_ERROR PASESession::HandlePBKDFParamResponse(const System::PacketBufferHandl
 exit:
     if (err != CHIP_NO_ERROR)
     {
-        SendErrorMsg(Spake2pErrorType::kUnexpected);
+        SendStatusReport(kProtocolCodeInvalidParam);
     }
     return err;
 }
 
 CHIP_ERROR PASESession::SendMsg1()
 {
+    size_t data_len                = EstimateTLVStructOverhead(kMAX_Point_Length, 1);
+    System::PacketBufferHandle msg = System::PacketBufferHandle::New(data_len);
+    VerifyOrReturnError(!msg.IsNull(), CHIP_ERROR_NO_MEMORY);
+
+    System::PacketBufferTLVWriter tlvWriter;
+    tlvWriter.Init(std::move(msg));
+
+    TLV::TLVType outerContainerType = TLV::kTLVType_NotSpecified;
+    ReturnErrorOnFailure(tlvWriter.StartContainer(TLV::AnonymousTag, TLV::kTLVType_Structure, outerContainerType));
+
     uint8_t X[kMAX_Point_Length];
     size_t X_len = sizeof(X);
 
     ReturnErrorOnFailure(
         mSpake2p.BeginProver(nullptr, 0, nullptr, 0, mPASEVerifier.mW0, kSpake2p_WS_Length, mPASEVerifier.mL, kSpake2p_WS_Length));
-
     ReturnErrorOnFailure(mSpake2p.ComputeRoundOne(NULL, 0, X, &X_len));
+    ReturnErrorOnFailure(tlvWriter.Put(TLV::ContextTag(1), ByteSpan(X, X_len)));
+    ReturnErrorOnFailure(tlvWriter.EndContainer(outerContainerType));
+    ReturnErrorOnFailure(tlvWriter.Finalize(&msg));
 
-    Encoding::LittleEndian::PacketBufferWriter bbuf(System::PacketBufferHandle::New(sizeof(uint16_t) + X_len));
-    VerifyOrReturnError(!bbuf.IsNull(), CHIP_ERROR_NO_MEMORY);
-    bbuf.Put16(GetLocalKeyId());
-    bbuf.Put(&X[0], X_len);
-    VerifyOrReturnError(bbuf.Fit(), CHIP_ERROR_NO_MEMORY);
-
-    mNextExpectedMsg = Protocols::SecureChannel::MsgType::PASE_Spake2p2;
+    mNextExpectedMsg = MsgType::PASE_Spake2p2;
 
     // Call delegate to send the Msg1 to peer
```

Comment:
Why delegate? Why not just "Send Pake1 to peer on the current exchange" ? Same comment applies to a lot of "stale" terminology in comments related to how messages are sent